regular_expressions/regular_expressions.md

```diff
@@ -0,0 +1,623 @@
+
+estimated_time_in_minutes: 60
+
+@pre_reqs
+Learners will be able to apply the skills learned in this module most rapidly if they have beginner-level experience in at least one query or programming language, such as SQL, R, Python, or Bash.  However, this module does not require coding experience and no computer code is written in this module.
+@end
+
+@learning_objectives  
+After completion of this module, learners will be able to:
+
+- Define an alphanumeric pattern in regex notation
+- Use capturing groups to extract data from a regular expression
+- Use anchors and boundaries in regular expressions
+- Test regular expressions in a browser
+@end
+
+@version_history 
+No previous versions.
+@end
+
+import: https://raw.githubusercontent.com/arcus/education_modules/main/_module_templates/macros.md
+-->
+
+# Regular Expressions
+
+@overview
+
+## What is a Regular Expression?
+
+The word *regular* (like *regulate*) comes from the Latin root for "rule": *regula*.  A **regular expression** is a specific way to **express** a **rule** for a pattern.  For example, one pattern rule might be for a phone number in the United States.  A phone number in the United States has:
+
+1. A three-digit area code, possibly enclosed by parentheses
+2. An optional separator, like a whitespace character (tab, space, etc.), hyphen, or a period
+3. A set of three digits
+4. An optional separator, like a whitespace character (tab, space, etc.), hyphen, or a period
+5. A set of four digits
+
+For example, these are all valid US phone numbers, and you might see them written various ways.
+
+* `(212) 555-1212`
+* `212.555.1212`
+* `212 555-1212`
+* `2125551212`
+* `(215)555.1212`
+
+How could you write an expression that expresses the rules we listed above?  This is the regular expression that captures the five rules we enumerated.
+
+`\(?\d{3}\)?[\s\-\.]?\d{3}[\s\-\.]?\d{4}`
+
+Using the pattern above, you could search for and find any phone number (assuming we got the rules right) in your data, without having to scan visually or search for specific area codes.  In this module, you'll learn how to construct regular expressions like the one above and how to use them in code.  For now, here's a sneak preview of what each segment of this regular expression mean:
+
+* `\(?` : Optionally, a single `(` character
+* `\d{3}`: Exactly three digits 0-9
+* `\)?` : Optionally, a single `)` character
+* `[\s\-\.]?`: Optionally, one of the following: a whitespace character, a hyphen, or a period.
+* `\d{3}`: Exactly three digits 0-9
+* `[\s\-\.]?`: Optionally, one of the following: a whitespace character, a hyphen, or a period.
+* `\d{4}`: Exactly four digits 0-9
+
+
+## Why Use Regex?
+
+There are two very common reasons you as a researcher might want to use regular expressions (also referred to as "regex", which you can pronounce either with "reg" rhyming with "beg" or "reg" rhyming with "wedge").
+
```

Comment:
Maybe add a comparison with a "normal" find function? And move this "why" section to the first thing learners see.